DuckDuckGo/SettingsViewController.swift

```diff
@@ -92,7 +96,10 @@ class SettingsViewController: UITableViewController {
     fileprivate lazy var featureFlagger = AppDependencyProvider.shared.featureFlagger
     fileprivate let syncService: DDGSyncing
     fileprivate let internalUserDecider: InternalUserDecider
-    private var internalStateChangeCancellable: Cancellable?
+#if NETWORK_PROTECTION
+    private let connectionObserver = ConnectionStatusObserverThroughSession()
```

Comment:
I'm glad to know these things helped reusability. :)

Please take the following feedback as a thought on possible future direction.  There's no need to do anything about it as the current implementation is fine.

I think we should consider eventually abstracting this and https://github.com/duckduckgo/iOS/pull/1861/files#diff-fde6768638fd5f87f67d6aa963d581eac0ccf5ffcb9abe259e9dd5143a2e65c8R378-R392 a bit more into a view model.